pkg/didcomm/common/service/service.go

```diff
@@ -21,6 +21,13 @@ type Handler interface {
 	HandleOutbound(msg *DIDCommMsg, myDID, theirDID string) error
 }
 
+// Headerable provides common methods for the message
+type Headerable interface {
+	GetID() string
```

Comment:
Simple accessors shouldn’t be prefixed with “Get”.

Also I think it should be a single interface method MsgHeader() or Header() that returns the Header struct.

Reply:
ok, MsgHeader is fine.

Reply:
Are you planning to directly use the header data fields directly or still call the accessor methods?

Reply:
I am going to use `MsgHeader()` function

Reply:
If the latter, I guess the multiple method interface could make sense with a better prefix.

Reply:
MsgID() ? MsgType?

Reply:
Msg as a prefix is okay.

Reply:
Are you ok with `Headerable` name?

Reply:
As you can see in my latter comments, I cannot tell how these methods will be used.

To your question, generally I would expect a `able` interface to have only one or two methods with the matching action. E.g., `Closeable` has a `Close()`. I would generally expect a `Headerable` to have a `Header()` or `MsgHeader()`.